### PR TITLE
Password generator policy enforcement

### PR DESCRIPTION
**Pull request by mportune-bw**
Enforcement of password generator policies.  Handling of UI widgets on generator screen could use some polish but I'd like a review for this so far.  (I don't see a way to comment on a review request, just a PR)

### CODE REVIEW

**Comment by kspearrin** [comment]
Did you format these files? I’m seeing a lot of inconsistencies on the if statement blocks.

**Comment by mportune-bw** [comment]
> Did you format these files? I’m seeing a lot of inconsistencies on the if statement blocks.

Fixed, forgot to hit the button.

File: src/Core/Services/PasswordGenerationService.cs [hunk 1]
```diff
@@ -255,7 +259,129 @@ public async Task<PasswordGenerationOptions> GetOptionsAsync()
...
+                }
+
+                if(enforcedOptions == null)
+                {
+                    enforcedOptions = new PasswordGeneratorPolicyOptions();
+                }
+
+                var currentPolicyMinLength = currentPolicy.Data["minLength"];
+                if(currentPolicyMinLength != null &&
+                   Convert.ToInt32(currentPolicyMinLength) > enforcedOptions.MinLength)
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: I think just casting to int would be enough here, and a little cleaner. `(int)currentPolicyMinLength`.

`Convert` is more for when _changing_ from one type to another, like converting a string `"1"` to an int `1`.

File: src/Core/Services/PasswordGenerationService.cs [hunk 2]
```diff
@@ -255,7 +259,129 @@ public async Task<PasswordGenerationOptions> GetOptionsAsync()
...
+                {
+                    enforcedOptions.MinLength = Convert.ToInt32(currentPolicyMinLength);
+                }
+
+                var currentPolicyUseUpper = currentPolicy.Data["useUpper"];
+                if(currentPolicyUseUpper != null && Convert.ToBoolean(currentPolicyUseUpper))
```
Comment (kspearrin) [inline src/Core/Services/PasswordGenerationService.cs]: Same with these booleans. They are already booleans, just happen to be represented as an object.

File: src/Core/Services/PasswordGenerationService.cs (hunk 1 above)
Comment (mportune-bw) [reply src/Core/Services/PasswordGenerationService.cs]: The object was coming back as an Int64 and throwing InvalidCastException.  Additional research shows double-unboxing will also work so I'll use that approach.

File: src/Core/Services/PasswordGenerationService.cs (hunk 2 above)
Comment (mportune-bw) [reply src/Core/Services/PasswordGenerationService.cs]: This was done for consistency - will change to simple cast

File: src/App/Pages/Generator/GeneratorPage.xaml [hunk 3]
```diff
@@ -152,8 +175,7 @@
                             HorizontalOptions="StartAndExpand" />
                         <Switch
                             IsToggled="{Binding Uppercase}"
-                            IsEnabled="{Binding EnforcedPolicyOptions.UseUppercase,
-                                Converter={StaticResource inverseBool}}"
+                            IsEnabled="{Binding IsUppercaseSwitchEnabled}"
```
Comment (kspearrin) [inline src/App/Pages/Generator/GeneratorPage.xaml]: @mportune-bw What prompted this change? This is how I thought you would have had to have done it fromt he beginning when I questioned it the first time.

Comment (mportune-bw) [reply src/App/Pages/Generator/GeneratorPage.xaml]: Hm I thought there was a scenario where this was necessary but now I'm wondering if I just confused myself with the styling of disabled vs off switch states.  I've reverted it locally and everything is still working as it should.  I'll submit a PR shortly.

**Pull request merged**